git.md

```diff
@@ -195,6 +195,26 @@ Fix nasty little bug ISSUE-51 #resolve #time 2h30m
 * Keep the history *clean* and *simple*. Before pushing any branch to the
   public always ensure that it conforms to the style-guide. If it doesn't:
   squash, rebase, reword commits as necessary.
+  
+* When merging multiple small commits into a single change, make sure to
+  keep high locality of the changes. In another words, if you're changing,
+  let's say, some elements of the UI, group changes
+  that occur in small vicinity of each other.
+
+* When merging multiple changes into one, make sure to _do not exceed
+  limit of 10 changes in one commit_. Additionally, always include
+  ticket/issue numbers in the commit log. For example:
+
+  ```git
+  Improve user-experience in subscription views PROJ-10
+
+  This batch includes:
+  * Tweak padding between buttons PROJ-13
+  * Ensure proper scrolling of the list-view PROJ-22
+  * Fix alpha layer on the toolbar PROJ-19
+  ```
+  
+  where `PROJ-10` is a super-ticket for all subsequent issues.
```

Comment:
Does this imply that it's valid only when PROJ-10 is a task, and PROJ-13/22/19 are subtasks, or can these issues be logically related, but not related in JIRA?

What I'm trying to say is that I feel the guide applies well to any batches of multiple changes, not necessarily strictly related by JIRA issues.

Reply:
I'm leaving here a little wiggle room. Sometimes there will be a supertask, and sometimes not (depends on the QA most of the time). The implicature of the guide is that if there are tickets, include them, and if not, consider making them. Ultimately, the choice is of the developer.